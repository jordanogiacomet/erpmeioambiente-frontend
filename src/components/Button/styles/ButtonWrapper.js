import styled, { css } from 'styled-components';

export const ButtonWrapper = styled.button`
  width: 100%;
  padding: 15px 0px;
  background-color: ${({ theme, backgroundColor }) => backgroundColor || theme.colors.secondary};
  color: ${({ theme, color }) => color || theme.colors.white};
  border: none;
  border-radius: 10px;
  font-size: ${({ theme, fontSize }) => fontSize || theme.font.sizes.large};
  font-weight: 700;
  cursor: pointer;
  transition: background-color 0.3s ease;

  &:hover {
    background-color: ${({ theme, hoverBackgroundColor }) => hoverBackgroundColor || theme.colors.hoverFormButton};
  }

  &:disabled {
    background-color: ${({ theme }) => theme.colors.disabled};
    cursor: not-allowed;
  }

  ${({ $variant }) =>
    $variant === 'forgot-password' &&
    css`
      width: fit-content;
      padding: 0;
      background-color: transparent;
      text-decoration: underline;
      white-space: nowrap;
      &:hover {
        background-color: transparent;
      }
    `}
`;