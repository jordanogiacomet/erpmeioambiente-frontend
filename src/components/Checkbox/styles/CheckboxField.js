import styled from 'styled-components';

export const CheckboxField = styled.input`
  width: auto;
  cursor: pointer;
`;