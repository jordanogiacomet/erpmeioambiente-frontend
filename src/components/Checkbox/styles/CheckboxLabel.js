import styled from 'styled-components';

export const CheckboxLabel = styled.label`
  margin-left: 8px;
  color: ${({ theme }) => theme.colors.secondary};
  font-size: ${({ theme }) => theme.font.sizes.large};
  font-weight: 600;
`;