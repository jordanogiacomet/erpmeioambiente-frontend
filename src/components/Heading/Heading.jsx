import P from 'prop-types';
import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.huge};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
  `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const Title = styled.h1`
  ${({ theme, $uppercase, $colorDark, size }) => css`
    color: ${($colorDark) ? theme.colors.primary : theme.colors.secondary};
    margin-bottom: 15px;
    ${titleSize[size](theme)};
    ${titleCase($uppercase)};
  `};
`;

export const Heading = ({ children, as = 'h1', size = 'big', uppercase = false, colorDark = false }) => {
  return (
    <Title as={as} $colorDark={colorDark} size={size} $uppercase={uppercase}>
      {children}
    </Title>
  );
};

Heading.propTypes = {
  children: P.node.isRequired,
  as: P.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
  size: P.oneOf(['small', 'medium', 'big']),
  uppercase: P.bool,
  colorDark: P.bool,
};


