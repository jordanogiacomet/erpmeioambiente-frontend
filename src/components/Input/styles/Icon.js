import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export const Icon = styled(FontAwesomeIcon)`
  position: absolute;
  right: 10px;
  top: 50%;
  transform: translateY(-50%);
  color: ${({ theme }) => theme.colors.secondary};
  pointer-events: none;  /* Prevent icon from capturing pointer events */
`;
