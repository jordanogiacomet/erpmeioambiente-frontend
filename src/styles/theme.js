// theme.js
export const theme = {
  colors: {
    primary: '#000000',    // Manter o azul para destaques e bot√µes
    secondary: '#FFFFFF',  // Manter o branco para fundos ou textos claros
    background: '#E0F2F1', // Verde claro para o fundo principal
    text: '#004D40',       // Verde escuro para o texto
    border: '#80CBC4',     // Verde para bordas sutis
    focus: '#004D40',      // Cor de foco, pode ser usada para bordas ao focar no input
  },
  font: {
    sizes: {
      small: '1rem',
      medium: '1.2rem',
      large: '1.6rem',
      xlarge: '2rem',
      huge: '2.4rem',
      xhuge: '3rem',
    },
  },
  media: {
    min: {
      sm: '(min-width: 640px)',
      md: '(min-width: 768px)',
      lg: '(min-width: 1024px)',
      xl: '(min-width: 1280px)',
      xxl: '(min-width: 1536px)',
    },
    max: {
      sm: '(max-width: 640px)',
      md: '(max-width: 768px)',
      lg: '(max-width: 1024px)',
      xl: '(max-width: 1280px)',
      xxl: '(max-width: 1536px)',
    }
  },
};
