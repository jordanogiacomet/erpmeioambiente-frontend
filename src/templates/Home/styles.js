import styled from 'styled-components';

export const Container = styled.section`
   background-color: black;
   display: flex;
   min-height: 100vh;
   align-items: center;
   justify-content: center;
`;
