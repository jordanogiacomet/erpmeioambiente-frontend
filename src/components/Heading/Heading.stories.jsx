import React from 'react';
import Heading from './Heading';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';

export default {
  title: 'Components/Heading',
  component: Heading,
  decorators: [(Story) => <ThemeProvider theme={theme}><Story /></ThemeProvider>],
  argTypes: {
    as: {
      control: {
        type: 'select',
        options: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'],
      },
    },
    size: {
      control: {
        type: 'select',
        options: ['small', 'medium', 'big'],
      },
    },
    uppercase: {
      control: {
        type: 'boolean',
      },
    },
    colorDark: {
      control: {
        type: 'boolean',
      },
    },
  },
};

const Template = (args) => <Heading {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Default Heading',
  size: 'big',
  as: 'h1',
};

export const Uppercase = Template.bind({});
Uppercase.args = {
  children: 'Uppercase Heading',
  uppercase: true,
};

export const DarkTheme = Template.bind({});
DarkTheme.args = {
  children: 'Dark Theme Heading',
  colorDark: true,
};


