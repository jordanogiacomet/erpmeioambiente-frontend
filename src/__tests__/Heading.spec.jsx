/* eslint-disable testing-library/no-node-access */
import React from 'react';
import renderTheme from '../styles/render-theme';
import 'jest-styled-components';
import Heading from '../components/Heading/Heading';
import { theme } from '../styles/theme';
import { screen } from '@testing-library/react';

describe('<Heading />', () => {
  it('should render with default props', () => {
    const { container } = renderTheme(<Heading>Test</Heading>);

    expect(container.firstChild).toHaveStyleRule('font-size', theme.font.sizes.xhuge);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render with uppercase text', () => {
    renderTheme(<Heading uppercase>texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle('text-transform: uppercase');
  });
  it('should render with different sizes', () => {
    renderTheme(
      <>
        <Heading size="small">small</Heading>
        <Heading size="medium">medium</Heading>
        <Heading size="big">big</Heading>
      </>
    );
    expect(screen.getByRole('heading', { name: 'small' })).toHaveStyle(`font-size: ${theme.font.sizes.xlarge}`);
    expect(screen.getByRole('heading', { name: 'medium' })).toHaveStyle(`font-size: ${theme.font.sizes.huge}`);
    expect(screen.getByRole('heading', { name: 'big' })).toHaveStyle(`font-size: ${theme.font.sizes.xhuge}`);
  });
  it('should render with color based on colorDark prop', () => {
    renderTheme(
      <>
        <Heading colorDark={false}>light</Heading>
        <Heading colorDark={true}>dark</Heading>
      </>
    );
    expect(screen.getByRole('heading', { name: 'light' })).toHaveStyle(`color: ${theme.colors.secondary}`);
    expect(screen.getByRole('heading', { name: 'dark' })).toHaveStyle(`color: ${theme.colors.primary}`);
  });
});