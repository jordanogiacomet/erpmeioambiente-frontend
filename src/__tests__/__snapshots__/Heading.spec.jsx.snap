// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render with default props 1`] = `
.c0 {
  color: #FFFFFF;
  margin-bottom: 15px;
  font-size: 3rem;
  text-transform: none;
}

<h1
  class="c0"
>
  Test
</h1>
`;
